src/turbolinks/visit.coffee

```diff
@@ -56,11 +56,12 @@ class Turbolinks.Visit
   loadCachedSnapshot: ->
     if snapshot = @getCachedSnapshot()
       isPreview = @shouldIssueRequest()
-      @render ->
-        @cacheSnapshot()
-        @controller.render({snapshot, isPreview}, @performScroll)
-        @adapter.visitRendered?(this)
-        @complete() unless isPreview
+      if snapshot.isPreviewable() or not isPreview
```

Comment:
I feel this new condition made the whole `loadCachedSnapshot ` function a quite harder to follow. Let me think if we can refactor this.